internal/databroker/server.go

```diff
@@ -125,52 +91,32 @@ func (srv *Server) UpdateConfig(options ...ServerOption) {
 	}
 	srv.cfg = cfg
 
-	for t, db := range srv.byType {
-		err := db.Close()
+	if srv.backend != nil {
+		err := srv.backend.Close()
 		if err != nil {
-			log.Warn().Err(err).Msg("databroker: error closing backend")
+			log.Error().Err(err).Msg("databroker: error closing backend")
 		}
-		delete(srv.byType, t)
+		srv.backend = nil
 	}
 
 	srv.initVersion()
 }
 
-// Delete deletes a record from the in-memory list.
-func (srv *Server) Delete(ctx context.Context, req *databroker.DeleteRequest) (*empty.Empty, error) {
-	_, span := trace.StartSpan(ctx, "databroker.grpc.Delete")
-	defer span.End()
-	srv.log.Info().
-		Str("type", req.GetType()).
-		Str("id", req.GetId()).
-		Msg("delete")
-
-	db, _, err := srv.getDB(req.GetType(), true)
-	if err != nil {
-		return nil, err
-	}
-
-	if err := db.Delete(ctx, req.GetId()); err != nil {
-		return nil, err
-	}
-
-	return new(empty.Empty), nil
-}
-
 // Get gets a record from the in-memory list.
 func (srv *Server) Get(ctx context.Context, req *databroker.GetRequest) (*databroker.GetResponse, error) {
 	_, span := trace.StartSpan(ctx, "databroker.grpc.Get")
 	defer span.End()
 	srv.log.Info().
+		Str("peer", grpcutil.GetPeerAddr(ctx)).
 		Str("type", req.GetType()).
 		Str("id", req.GetId()).
 		Msg("get")
 
-	db, _, err := srv.getDB(req.GetType(), true)
+	db, _, err := srv.getBackend(true)
 	if err != nil {
 		return nil, err
 	}
-	record, err := db.Get(ctx, req.GetId())
+	record, err := db.Get(ctx, req.GetType(), req.GetId())
 	if err != nil {
 		return nil, status.Error(codes.NotFound, "record not found")
```

Comment:
I think errors need be mapped - `storage.ErrNotFound => codes.NotFound`, everything else to `codes.InternalError`